pkgs/development/python-modules/disnake/default.nix

```
@@ -0,0 +1,71 @@
+{
+  lib,
+  setuptools,
+  aiohttp,
+  buildPythonPackage,
+  fetchFromGitHub,
+  libopus,
+  pynacl,
+  pythonOlder,
+  withVoice ? true,
+  ffmpeg,
+}:
+
+buildPythonPackage rec {
+  pname = "disnake";
+  version = "2.10.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "DisnakeDev";
+    repo = "disnake";
+    tag = "v${version}";
+    hash = "sha256-MQxYkUA3uclmY2cKBr4DsBg79ovsH1EsMOjiVPGaLVE=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies =
+    [
+      aiohttp
+    ]
+    ++ lib.optionals withVoice [
+      libopus
+      pynacl
+      ffmpeg
+    ];
+
+  postPatch =
+    ''
+      substituteInPlace "disnake/opus.py" \
+        --replace-fail "ctypes.util.find_library("opus")" "'${libopus}/lib/libopus.so.0'"
```

This has bash quoting issues.
```suggestion
        --replace-fail 'ctypes.util.find_library("opus")' "'${libopus}/lib/libopus.so.0'"
```

---

I tested this locally and it builds both with and without `withVoice`.

---

Decided to move the top part into `withVoice` to hopefully reduce dependencies, also fixed the shell quoting issue